include/micm/solver/state.hpp

```diff
@@ -8,6 +8,7 @@
 #include <micm/util/jacobian.hpp>
 #include <micm/util/matrix.hpp>
 #include <micm/util/sparse_matrix.hpp>
+#include <micm/util/cuda_dense_matrix.hpp>
```

Comment:
@mattldawson @mwaxmonsky @K20shores because I add the `cuda_dense_matrix.hpp` header here, all the CI tests will now try to include the CUDA stuff. I think this is the reason most CI tests fail as those containers do not install CUDA. Is there a better way to handle the issue here?

Reply:
You could extend this class with a `CudaState` class that implements the synching functions. You would probably also need to modify the `Solver` class to return the correct type of state class from the `GetState` function.

Reply:
Thanks @mattldawson . I have extended the `state` class to `cudastate` class at https://github.com/NCAR/micm/pull/579/commits/cd5735edde54872fb4e2a27ac5a278c73576c8d4 (hopefully I did it correctly) but I don't know how to revise the `Solver` class to get the `cudastate` for the CUDA tests. Can you provide more details here?

Reply:
We'll need to figure out how to make the solver builder return the appropriate state

https://github.com/NCAR/micm/blob/85232c5e80f1b0b3c06b6aa236c08e588b86ba2a/include/micm/solver/solver_builder.inl#L300-L307

Reply:
One way would be to add another function onto the builder, `BuildForGPU` that will do the same things as in this function, but return a `Solver<SolverPolicy, CudaState<DenseMatrixPolicy, SparseMatrixPolicy>>`

Reply:
Thanks @mattldawson and @K20shores for your help. I have added the `CudaState` class and updated the cuda solver builder accordingly.